### PR TITLE
VZ-8502 VZ Tools Diagnosis on Cpu Limits

### PR DESCRIPTION
**Pull request by absBitm**
This adds the diagnosis of issues related to cpu limits into the nodes

### CODE REVIEW

File: tools/vz/pkg/analysis/internal/util/cluster/pods.go [hunk 1]
```diff
@@ -216,25 +216,36 @@ func handleImagePullBackOff(log *zap.SugaredLogger, clusterRoot string, podFile
...
+		issueReporter.AddKnownIssueMessagesFiles(report.InsufficientMemory, clusterRoot, messages[report.InsufficientMemory], files)
+	}
+	if len(messages[report.InsufficientCpu]) > 0 {
+		issueReporter.AddKnownIssueMessagesFiles(report.InsufficientCpu, clusterRoot, messages[report.InsufficientCpu], files)
+	}
+}
+
 func podStatusConditionIssues(log *zap.SugaredLogger, clusterRoot string, podFile string, pod corev1.Pod, issueReporter *report.IssueReporter) (err error) {
-	log.Debugf("MemoryIssues called for %s", clusterRoot)
+	log.Debugf("Memory or Cpu Issues called for %s", clusterRoot)
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/cluster/pods.go:229]: Please use CPU in anything we print to the standard out or report

Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/pods.go:229]: ok thanks

File: tools/vz/pkg/analysis/internal/util/report/issue.go [hunk 2]
```diff
@@ -128,6 +129,7 @@ var knownIssues = map[string]Issue{
 	ImagePullNotFound:         {Type: ImagePullNotFound, Summary: "Failure(s) pulling images have been detected due to the image not being found", Informational: false, Impact: 10, Confidence: 10, Actions: []Action{KnownActions[ImagePullNotFound]}},
 	ImagePullService:          {Type: ImagePullService, Summary: "Failure(s) pulling images have been detected due to the service not being available, the service may be unreachable or may be incorrectly specified", Informational: false, Impact: 10, Confidence: 10, Actions: []Action{KnownActions[ImagePullService]}},
 	InsufficientMemory:        {Type: InsufficientMemory, Summary: "Failure(s) due to insufficient memory on nodes have been detected", Informational: false, Impact: 10, Confidence: 10, Actions: []Action{KnownActions[InsufficientMemory]}},
+	InsufficientCPU:           {Type: InsufficientCPU, Summary: "Failure(s) due to insufficient cpu on nodes have been detected", Informational: false, Impact: 10, Confidence: 10, Actions: []Action{KnownActions[InsufficientCPU]}},
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/report/issue.go:132]: CPU instead of cpu

File: tools/vz/pkg/analysis/internal/util/cluster/pods.go [hunk 3]
```diff
@@ -216,25 +216,36 @@ func handleImagePullBackOff(log *zap.SugaredLogger, clusterRoot string, podFile
...
 	if len(pod.Status.Conditions) > 0 {
-		messages := make([]string, 0)
+		messages := make(map[string][]string)
 		for _, condition := range pod.Status.Conditions {
 			if strings.Contains(condition.Message, "Insufficient memory") {
-				messages = append(messages, fmt.Sprintf("Namespace %s, Pod %s, Status %s, Reason %s, Message %s",
+				messages[report.InsufficientMemory] = append(messages[report.InsufficientMemory], fmt.Sprintf("Namespace %s, Pod %s, Status %s, Reason %s, Message %s",
+					pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, condition.Status, condition.Reason, condition.Message))
+			}
+			if strings.Contains(condition.Message, "Insufficient cpu") {
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/cluster/pods.go:238]: Is "Insufficient cpu" is from pods.json ?

Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/pods.go:238]: it reads the pod's condition status and search the text inside it

File: tools/vz/pkg/analysis/internal/util/cluster/pods.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2021, 2022, Oracle and/or its affiliates.
+// Copyright (c) 2023, Oracle and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/pods.go:1]: it was throwing the error on pipeline

File: tools/vz/pkg/analysis/internal/util/report/issue.go (hunk 2 above)
Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/report/issue.go]: done

File: tools/vz/pkg/analysis/internal/util/cluster/pods.go (hunk 3 above)
Comment (bhatpmk) [reply tools/vz/pkg/analysis/internal/util/cluster/pods.go:238]: Thank you. For some of these cases, probably it is a good idea to use case-insensitive comparison

**Review by bhatpmk — APPROVED** [review]
LGTM

**Pull request merged**